### ISSUE TITLE
System.NotSupportedException

### ISSUE THREAD

**Issue by biggert** [mkoertgen/hello.ocr#2]
Which one of the quicklinks addresses this exception? Did I miss something?

Unhandled Exception: System.TypeLoadException: Could not find Windows Runtime ty
pe 'WindowsPreview.Media.Ocr.OcrResult'. ---> System.NotSupportedException: Type
s from custom Windows Runtime components are not supported in desktop applicatio
ns.
   --- End of inner exception stack trace ---
   at hello.ocr.Program.Main(String[] args)

**Comment by biggert** [comment]
I see I was missing the steps in "Update #2" from http://stackoverflow.com/questions/27198656/how-to-use-microsoft-ocr-library-microsoft-windows-ocr-in-an-asp-net-mvc4-we. I'll close this as I have yet to try that.

**Comment by mkoertgen** [comment]
Honestly, i'm stuck on the same thing and on it. The sample referred in the SO answer overcomes this limitation by 

1. have a web api rest server (classic)
2. talking to a winrt app 

While this definitely works i think this awkward. Why made Microsoft it so hard to use this? I understand this limitation has no strong (technical) reasons but is just arbitrary.

**Comment by PaunCostin** [comment]
Hey,

I downloaded the code and wanted to run. Same issue as you guys had above.
Was there any progress on this ?

Were you ever able to run the code ? And if so, how ?

I would really appreciate you answer.

Thanks,
Costin

**Comment by mkoertgen** [comment]
Sadly not. It seems that `Windows.Media.Ocr` can only be used inside WinRT App.

**Comment by PaunCostin** [comment]
Hi and thanks for the reply,

I succeeded in using `Windows.Media.Ocr` in a "classic" desktop app on windows 10 with VS 2015.
I used `UwpDesktop ` NuGet package (by Vladimir Postel) to create the necessary bridge. 
For being able to use the OCR library I added a reference to `Windows.Foundation.UniversalApiContract.winmd`.

Of course it works just on Win 10 because I don't really know how (or if it's even possible) to isolate the required dll's in order to pack them with the program.
Maybe the code will also be runable on Windows Server 2016 for the ones in need of a server service.

Some sample C# code for Windows.Media.OCR can be found here:
https://github.com/Microsoft/Windows-universal-samples/tree/master/Samples/OCR/cs

Costin
